File: ruby/json/extract_title_from_json.rb
# -*- coding: utf-8 -*-

require 'json'
require 'awesome_print'

class Extractor
  def initialize(args)
    @filename = args.shift || "json.txt"
  end

  def get_title
    arr = Array.new
    open(@filename) do |file|
      file.each_line do |line|
        key, tag, json = line.force_encoding("utf-8").strip.split("\t")
        JSON.parse(json).each {|key, value|
          arr << value if key == 'title'
        }
      end
    end

    arr
  end
end

if __FILE__ == $0
  extractor = Extractor.new(ARGV)
  result = extractor.get_title
  ap result
end

File: python/json/extract_title_from_json.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import sys
import json

class Extractor:

    def __init__(self, args):
        self.filename = args[1]

    def get_title(self):
        arr = []
        file = open(self.filename, 'r')
        for line in file:
            key, tag, value = line.rstrip().split("\t")
            json_obj = json.loads(value)
            for key, value in json_obj.items():
                if key == 'title':
                    arr.append(value)

        file.close
        return arr

    def __output(self, key, tag, value):
        print(key, tag, value, sep="\t")

if __name__ == '__main__':
    argsmin = 1
    version = (3, 0)
    if sys.version_info > (version):
        if len(sys.argv) > argsmin:
            extractor = Extractor(sys.argv)
            result = extractor.get_title()
            print(result)
        else:
            print("This program needs at least %(argsmin)s arguments" %
                  locals())
    else:
        print("This program requires python > %(version)s" % locals())
